File: domain/src/main/java/io/github/PrzeBarCore/Receipt/Receipt.java
package io.github.PrzeBarCore.Receipt;

import io.github.PrzeBarCore.ValueObjects.Company;
import io.github.PrzeBarCore.ValueObjects.MonetaryAmount;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

class Receipt {
    private Integer id;
    private LocalDateTime issuedOnDateTime;
    private MonetaryAmount totalValue;
    private boolean isContainingListOfItems;
    private List<ReceiptItem> items;

    class ReceiptItem {
        private Integer id;
        private String name;
        private Integer productId;
        private Double quantity;
        private MonetaryAmount regularPrice;
        private MonetaryAmount discount;
        //transactionCategory
        private Integer productCategory;
    }
}

File: front/finances-tracking-system/src/app/receipt/model/receipt.ts
import { Category } from 'src/app/category/model/category';
import { Product } from 'src/app/product/model/product';

export class Receipt {
  constructor(
    public id: number,
    public totalDiscount: number,
    public items: ReceiptItem[]
  ) {}
}

export class ReceiptItem {
  public isEdited = false;
  constructor(
    public id: number,
    public name: string,
    public quantity: number,
    public regularPrice: number,
    public discount: number,
    public category: Category,
    public product: Product
  ) {}
}
